### ISSUE TITLE
[improvement] GPUTrail 2D variant

### ISSUE THREAD

**Issue by francogp** [celyk/GPUTrail#2]
Could this be adapted to a 2D version, or would it require a complete rewrite? I can try, but I need a confirmation.

**Comment by celyk** [comment]
Yes I think so. I initially planned to make a 2D version, and I avoided features that aren't available in 2D or Godot 3.x. You might find it easier to branch from https://github.com/celyk/GPUTrail/commit/3c1926b87d5fd703dff975953e59d2a4cbfea597 since it's cleaner

You'll have to go in each file and change all the 3D stuff to 2D stuff. I wish there was a unified way to support both. Just throw away the Z component of the trail! You *can* use a `SubViewport` to get the trail into 2D as well.

**Comment by Talent-Lost** [comment]
Did you ever adapt this to 2d? I'd love to see what that looks like

**Comment by francogp** [comment]
> Did you ever adapt this to 2d? I'd love to see what that looks like

@celyk I created a prototype, in a branch. If you want to copy and merge the concept in your project, it needs some polish (its more a common game project than a plugin).

https://github.com/francogp/GPUTrail

I resume all my changes in
https://github.com/francogp/GPUTrail/blob/main/README_2D.md
